include/vcpkg/base/jsonreader.h

```diff
@@ -161,29 +171,27 @@ namespace vcpkg::Json
         template<class Type>
         Optional<std::vector<Type>> array_elements(const Array& arr, IDeserializer<Type>& visitor)
         {
-            std::vector<Type> result;
-            m_path.emplace_back();
+            Optional<std::vector<Type>> result{std::vector<Type>()};
+            PathGuard guard{m_path};
             for (size_t i = 0; i < arr.size(); ++i)
             {
                 m_path.back().index = static_cast<int64_t>(i);
                 auto opt = visitor.visit(*this, arr[i]);
-                if (auto p = opt.get())
+                if (auto parsed = opt.get())
                 {
-                    result.push_back(std::move(*p));
+                    if (auto result_vec = result.get())
+                    {
+                        result_vec->push_back(std::move(*parsed));
+                    }
                 }
                 else
                 {
                     this->add_expected_type_error(visitor.type_name());
-                    for (++i; i < arr.size(); ++i)
-                    {
-                        m_path.back().index = static_cast<int64_t>(i);
-                        auto opt2 = visitor.visit(*this, arr[i]);
-                        if (!opt2) this->add_expected_type_error(visitor.type_name());
-                    }
+                    result.clear();
```

Comment:
This removes visiting the other array elements, which prevents us from providing additional independent error messages for the users.

Reply:
No, it doesn't do that. We always visit all the elements, we just don't return results for them.

Reply:
Ahhh, sorry for not reading closely enough. LGTM.